### PR TITLE
squash pkg/spiffs: (opendir, readdir, closedir) Use pointer to private data directly instead of copying

### PR DESCRIPTION
**Pull request by jnohlgard**

### CODE REVIEW

File: pkg/spiffs/fs/spiffs_fs.c [hunk 1]
```diff
@@ -260,31 +260,30 @@ static int _fstat(vfs_file_t *filp, struct stat *buf)
...
+    spiffs_DIR *d = (spiffs_DIR *)&dirp->private_data.buffer[0];
     (void) abs_path;
 
     if (VFS_DIR_BUFFER_SIZE < sizeof(spiffs_DIR)) {
         return -EOVERFLOW;
     }
 
-    SPIFFS_opendir(&fs_desc->fs, dirname, &d);
-    memcpy(dirp->private_data.buffer, &d, sizeof(d));
+    spiffs_DIR *res = SPIFFS_opendir(&fs_desc->fs, dirname, d);
```
Comment (jnohlgard) [inline pkg/spiffs/fs/spiffs_fs.c:270]: Also added a check on the return value of opendir. It fails if the fs is not mounted or if `SPIFFS_CHECK_CFG` fails.

**Comment by vincent-d** [comment]
@gebart , are we sure the memory alignment is OK. That was why I used a memcpy.

**Comment by jnohlgard** [comment]
The alignment of private_data.buffer, you mean? The alignment of that should be the same as for `void *` since it's part of a union with that type, but maybe we should change the buffer to be of a different data type?

**Comment by vincent-d** [comment]
yep I meant private_data.buffer, but I've missed the union thing, so that should be OK indeed.

**Pull request merged**